### DISCUSSION TITLE
Sharing variables between nodes

### DISCUSSION THREAD

**Discussion by sofyanajridi** [microsoft/promptflow#3533] [Q&A]
Hello,

I wanted to know if it is possible to share variables between nodes. E.g lets say I have an initial flow that sets up a AzureOpenAI connection (not using the existing LLM node):

Node 1:
```
client = AzureOpenAI(....)
r = client.embeddings.create(...)
```

Now for Node 2, I would like to reuse the same client from Node 1. Is this possible or should I always have to redefine the variable?

If this is not possible, I would think a feature request like the following could be interesting?

Node 1:
```
client = AzureOpenAI(....)
global_config["client"] = client

r = client.embeddings.create(...)
```

Node 2:
```
client =global_config["client"] 
r = client.embeddings.create(...)
```

**Comment by D-W-** [comment]
Hi @sofyanajridi , currently it's not supported in DAG flow. But you can try our [class-based flex flow](https://microsoft.github.io/promptflow/how-to-guides/develop-a-flex-flow/class-based-flow.html) and define your `AzureOpenAI` client inside `__init__` method, each line run can share same client to avoid create duplicate clients.

**Reply by sofyanajridi** [reply]
Thank you, will have a look!
